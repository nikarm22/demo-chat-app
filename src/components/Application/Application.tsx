import React from 'react';
import Grid from '@material-ui/core/Grid';
import { FriendsList } from '../FriendsList/FriendsList';
import { makeStyles } from '@material-ui/core';
import { ChatBody } from '../ChatBody/ChatBody';
import { Route } from 'react-router-dom';

const useStyles = makeStyles(() => ({
  root: {
    height: '100%',
  },
  items: {
    height: '100%',
  },
}));

function App() {
  const { root, items } = useStyles();
  return (
    <Grid container className={root}>
      <Grid item xs={3} className={items}>
        <FriendsList />
      </Grid>
      <Route path='/:id'>
        <Grid item xs={9} className={items}>
          <ChatBody />
        </Grid>
      </Route>
    </Grid>
  );
}

export default App;
