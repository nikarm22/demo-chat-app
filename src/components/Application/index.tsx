import React from 'react';
import Grid from '@material-ui/core/Grid';
import { FriendsList } from '../FriendsList';
import { IFriend } from '../FriendsList/types';

const mockFriends = Array(10).fill(null).map((_, i) => ({ id: i, name: Math.random() > .5 ? `Valod N${i}` : `Gugo N${i}`, avatar: Math.random() > .5 ? 'https://via.placeholder.com/150x150' : null } as IFriend))

function App() {
  return (
    <Grid container>
      <Grid item xs={3}>
        <FriendsList friends={mockFriends} activeId={4} />
      </Grid>
      <Grid item xs={9}>
asdafsdf
      </Grid>
    </Grid>
  );
}

export default App;
