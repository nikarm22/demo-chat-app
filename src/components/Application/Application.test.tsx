import React from 'react';
import { render, act } from '@testing-library/react';
import Application from './Application';
import { MemoryRouter } from 'react-router-dom';

describe('<Application />', () => {
  jest.mock('../../services/friends', () => {
    return {
      getAllFriends: jest.fn().mockResolvedValue([]),
    };
  });

  it('should render the component without errors', async () => {
    await act(async () => {
      const { asFragment } = render(
        <MemoryRouter initialEntries={['/']}>
          <Application />
        </MemoryRouter>
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
