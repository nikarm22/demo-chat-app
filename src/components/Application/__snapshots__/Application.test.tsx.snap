// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Application /> should render the component without errors 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root makeStyles-items-2 MuiGrid-item MuiGrid-grid-xs-3"
    >
      <ul
        class="MuiList-root makeStyles-root-3 MuiList-padding"
      />
    </div>
  </div>
</DocumentFragment>
`;
